src/main/java/ro/societateahermes/backendservice/entities/DTO/MySubmissionDTO.java

```diff
@@ -0,0 +1,44 @@
+package ro.societateahermes.backendservice.entities.DTO;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ro.societateahermes.backendservice.entities.form.Answer;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class MySubmissionDTO {
+
+    //fields related to the user
+    @NotBlank(message = "Email is mandatory")
+    @Pattern(regexp = "^\\w+([-+.']\\w+)*@\\w+([-.]\\w+)*\\.\\w+([-.]\\w+)*$", message = "Email invalid!")
+    private String email;
+    @NotBlank(message = "Username is mandatory")
+    private String username;
+    @NotBlank(message = "Firstname is mandatory")
+    @Pattern(regexp = "[a-zA-Z]+", message = "Firstname cannot contain numbers")
+    private String firstName;
+    @NotBlank(message = "Lastname is mandatory")
+    @Pattern(regexp = "[a-zA-Z]+", message = "Lastname cannot contain numbers")
+    private String lastName;
+    @NotBlank(message = "University is mandatory")
+    private String university;
+    @NotBlank(message = "Password is mandatory")
+    private String password;
```

Comment:
Un Dto nu trebuie sa contina neaparat toate fieldurile entitatii pe care o reprezinta. Ce e obligatoriu in schimb este id-ul acelei entitati. Aici spre exemplu parola ar putea fi eliminata din dto daca nu e necesara mai tarziu pentru ceva. In schimb, trebuie adaugat un id

Reply:
Userul mai contine si anul de studiu,specializarea si limba de studiu....stiu ca am stabilit la inceput ca campurile ID, email, username, firstName, lastName, university, password sa fie continute de toti userii, si celelate sa fie adaugate la un eventual update( cand isi completeaza profilul sau ceva). Asa ca m-am gandit ca in form apar ca intrebari toate astea si atunci vin in dto. Iar id-ul userului se genereaza la salvarea in db si nu am considerat ca e necesar ca field in Dto.

Reply:
Atunci da. Ai dreptate si e foarte bine cum faci. Ai numai grija ca mereu cand trimiti date catre frontend sa nu trimiti niciodata parola si sa trimiti mereu id-ul. Dto-ul asta e bun ca sa primesti date, dar nu ca sa si trimiti. Ar trebui sa ai dto-uri separate pentru fiecare functionalitate care cere date despre o entitate anume